nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -219,15 +221,41 @@ public void createNamespace(Namespace namespace, Map<String, String> metadata) {
     checkNamespaceIsValid(namespace);
     getRef().checkMutable();
     ContentKey key = ContentKey.of(namespace.levels());
-    org.projectnessie.model.Namespace content =
-        org.projectnessie.model.Namespace.of(key.getElements(), metadata);
     try {
       Content existing = api.getContent().reference(getReference()).key(key).get().get(key);
       if (existing != null) {
         throw namespaceAlreadyExists(key, existing, null);
       }
+
+      // check if the parent namespace exists
+      List<ContentKey> keys = Lists.newArrayList();
+      Map<ContentKey, Content> contentInfo = Maps.newHashMap();
+      for (int i = 0; i < namespace.levels().length; i++) {
+        Namespace parent = Namespace.of(Arrays.copyOf(namespace.levels(), i + 1));
+        ContentKey parentKey = ContentKey.of(parent.levels());
+        keys.add(parentKey);
+        Content parentContent =
+            api.getContent().reference(getReference()).key(parentKey).get().get(parentKey);
```

Comment:
It's better to do only one call to Nessie for all required content-keys - including the one in line 225.